src/openms/source/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.cpp

```diff
@@ -0,0 +1,291 @@
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Julia Thueringer $
+// $Authors: Julia Thueringer $
+// --------------------------------------------------------------------------
+
+#include <OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.h>
+// calculate pearson distance
+#include <OpenMS/MATH/STATISTICS/StatisticFunctions.h>
+// create binary tree
+#include <OpenMS/DATASTRUCTURES/DistanceMatrix.h>
+#include <OpenMS/DATASTRUCTURES/BinaryTreeNode.h>
+#include <OpenMS/COMPARISON/CLUSTERING/ClusterHierarchical.h>
+#include <include/OpenMS/COMPARISON/CLUSTERING/AverageLinkage.h>
+// align maps and generate output
+#include <OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentTransformer.h>
+#include <OpenMS/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmIdentification.h>
+
+#include <include/OpenMS/APPLICATIONS/MapAlignerBase.h>
+
+using namespace std;
+
+namespace OpenMS
+{
+
+  MapAlignmentAlgorithmTreeGuided::MapAlignmentAlgorithmTreeGuided() :
+          DefaultParamHandler("MapAlignmentAlgorithmTreeGuided"),
+          ProgressLogger()
+  {
+    defaults_.insert("model:", TOPPMapAlignerBase::getModelDefaults("b_spline"));
+    defaults_.setValue("model_type", "b_spline", "Options to control the modeling of retention time transformations from data");
+    defaults_.setValidStrings("model_type", {"linear","b_spline","lowess","interpolated"});
+    defaults_.insert("align_algorithm:", MapAlignmentAlgorithmIdentification().getDefaults());
+    defaults_.setValue("align_algorithm:use_feature_rt", "true", "Because only aligning feature maps with this algorithm, use the retention time of the centroid of the feature (apex of the elution profile) that the peptide was matched to. If different identifications are matched to one feature, only the peptide closest to the centroid in RT is used.#br#Precludes &apos;use_unassigned_peptides&apos;.");
+    defaults_.setValidStrings("align_algorithm:use_feature_rt", {"true"});
+    defaultsToParam_();
+  }
+
+  MapAlignmentAlgorithmTreeGuided::~MapAlignmentAlgorithmTreeGuided()
+  {
+  }
+
+  void MapAlignmentAlgorithmTreeGuided::updateMembers_()
+  {
+    align_algorithm_.setParameters(param_.copy("align_algorithm:", true));
```

Comment:
hmm I think if the user changes the model_type, then the defaults of model: also need to be updated?

Reply:
maybe we need to extract the defaults for all models here

Reply:
Can you explain what you mean? I think by overwriting 'model_param_' with 'model_param_.copy(model_type_+":", true);' the user input is used (because the trafoXML output contains the changed parameters). Am I missing something?

Reply:
Sorry, my fault! You are right. I'll change it.